pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -322,20 +319,19 @@ def on_pretrain_routine_start(self, trainer, pl_module):
 
     @rank_zero_only
     def on_validation_end(self, trainer, pl_module):
-        # only run on main process
-        if trainer.global_rank != 0:
-            return
-
-        # no models are saved
-        if self.save_top_k == 0:
-            return
-
-        if trainer.running_sanity_check:
-            return
-
         metrics = trainer.logger_connector.callback_metrics
         epoch = trainer.current_epoch
 
+        if (
+            trainer.global_rank != 0  # only run on main process
+            or self.save_top_k == 0  # no models are saved
+            or self.period < 1  # no models are saved
```

Comment:
not that this will remove a usecase with val checking multiple times per epoch:
https://github.com/PyTorchLightning/pytorch-lightning/issues/3646
which was possible before by setting period=0, but perhaps we should find another way for that. 
Just making this comment so we are aware of it.

Reply:
yep, using `period = 0` to do it means relying on an implementation detail, so a bad idea.

Reply:
yes, cool. let's figure out how we can support it in a proper way.

Reply:
I mean different PR :)